### DISCUSSION TITLE
How to add a block programmatically instead of dragging using the block ID?

### DISCUSSION THREAD

**Discussion by gitpugal** [GrapesJS/grapesjs#5794] [Q&A]
I want to add a block using it's block id using some function instead of dragging the block and adding it to the canvas.

**Comment by artf** [comment]
[Get](https://grapesjs.com/docs/api/block_manager.html#get) the block by id, then [append](https://grapesjs.com/docs/api/component.html#append) its [content](https://grapesjs.com/docs/api/block.html#getcontent) to a component.
